cli/pcluster/config_sanity.py

```diff
@@ -530,6 +530,56 @@ def validate(self, resource_type, resource_value):  # noqa: C901 FIXME
         # FSX FS Id check
         elif resource_type in ["fsx_fs_id", "FSx_storage_capacity", "FSx_imported_file_chunk_size", "FSx_export_path"]:
             self.__validate_fsx_parameters(resource_type, resource_value)
+        elif resource_type == "EFA":
+            valid_instances = get_supported_efa_instances(self.region)
+            if "ComputeInstanceType" in resource_value:
+                if resource_value.get("ComputeInstanceType") not in valid_instances:
+                    self.__fail(resource_type, "Compute Instance needs to be one of %s" % valid_instances)
+            else:
+                self.__fail(resource_type, "Compute Instance needs to be one of %s" % valid_instances)
+            if resource_value.get("PlacementGroup", "NONE") == "NONE":
+                self.__fail(resource_type, "Placement group is required, set placement_group.")
+            if "VPCSecurityGroupId" in resource_value:
+                sg_id = resource_value.get("VPCSecurityGroupId")
+                try:
+                    ec2 = boto3.client(
+                        "ec2",
+                        region_name=self.region,
+                        aws_access_key_id=self.aws_access_key_id,
+                        aws_secret_access_key=self.aws_secret_access_key,
+                    )
+                    sg = ec2.describe_security_groups(GroupIds=[sg_id]).get("SecurityGroups")[0]
+                    in_rules = sg.get("IpPermissions")
+                    out_rules = sg.get("IpPermissionsEgress")
+
+                    allowed_in = False
+                    allowed_out = False
+                    for rule in in_rules:
+                        if (
+                            rule.get("IpProtocol") == "-1"
+                            and len(rule.get("UserIdGroupPairs")) > 0
+                            and rule.get("UserIdGroupPairs")[0].get("GroupId") == sg_id
```

Comment:
Can this have more than one GroupPair?

Reply:
Here's what the `rule` object looks like:

```
{u'PrefixListIds': [], u'FromPort': 0, u'IpRanges': [], u'ToPort': 65535, u'IpProtocol': 'tcp', u'UserIdGroupPairs': [{u'UserId': '822857487308', u'GroupId': 'sg-00e715cfdebcb97bb'}], u'Ipv6Ranges': []}
{u'PrefixListIds': [], u'FromPort': 22, u'IpRanges': [{u'CidrIp': '0.0.0.0/0'}], u'ToPort': 22, u'IpProtocol': 'tcp', u'UserIdGroupPairs': [], u'Ipv6Ranges': []}
```

I've done some testing to see if I can extend `UserIdGroupPairs` to make it longer than 1. I can't so I've added a comment explaining this behavior.